File: split/popalignment.pas
procedure popalignment;
var
  p: halfword;
begin
  begin
    mem[curhead].hh.rh := avail;
    avail := curhead;
    {dynused:=dynused-1;}
end;
  p := alignptr;
  curtail := mem[p + 4].hh.rh;
  curhead := mem[p + 4].hh.lh;
  alignstate := mem[p + 3].int;
  curloop := mem[p + 2].int;
  curspan := mem[p + 1].hh.rh;
  mem[29992].hh.rh := mem[p + 1].hh.lh;
  curalign := mem[p].hh.lh;
  alignptr := mem[p].hh.rh;
  freenode(p, 5);
end;

File: cpp/popalignment.cpp
#include "popalignment.h"
#include "freenode.h"

void popalignment(void)
{
	link(curhead) = avail;
	avail = curhead;
	auto p = alignptr;
	curtail = link(p+4);
	curhead = info(p+4);
	alignstate = mem[p+3].int_;
	curloop = mem[p+2].int_;
	curspan = rlink(p);
	preamble() = llink(p);
	curalign = info(p);
	alignptr = link(p);
	freenode(p, 5);
}
